### PR TITLE
docs: Reviewed and tweaked wording in README.md

### PR DESCRIPTION
**Pull request by hsuamz**
### Summary

Documentation review and changes for the upcoming 0.4.0 release.

### Description

- Reviewed `Connection Plugin Manager` and `Enhanced Failure Monitoring` sections.
- Updated wording for the general README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -284,47 +282,48 @@ public class FailoverSampleApp2 {
...
 
 ### Connection Plugin Manager
-Connection plugin manager initializes, triggers, and cleans up a chain of connection plugins. Connection plugins are widgets attached to each `Connection` objects to help execute additional or supplementary logic related to that `Connection`. [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc#enhanced-failure-monitoring) is one example of a connection plugin. 
+The connection plugin manager initializes, triggers, and cleans up a chain of connection plugins. Connection plugins are widgets attached to each `Connection` object to help execute additional or supplementary logic related to that `Connection`. [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc#enhanced-failure-monitoring) is one example of a connection plugin.
```
Comment (karenc-bq) [inline README.md:288]: The more I look at it the more awkward this line looks. 
> Enhanced Failure Monitoring is one example of a connection plugin.

I think we should refer to the feature as Enhanced Failure Monitoring and refer the plugin as Enhanced Failure Monitoring Connection Plugin

Comment (susanmdouglas-aws) [reply README.md:288]: Would it be better phrased:

Enhanced Failure Monitoring is just one example of a connection plugin; you can use the 'Enhanced Failure Monitoring' Connection plugin to implement the functionality.

Comment (karenc-bq) [reply README.md:288]: I propose we rename `Enhanced Failure Monitoring` to `Enhanced Failure Monitoring Plugin`.

Comment (hsuamz) [reply README.md:288]: Great, let's do that name change. I'll merge this PR in and we can open up another PR that does the renaming.

**Pull request merged**